components/agreements/AgreementForm.tsx

```diff
@@ -27,6 +27,23 @@ import { AGREEMENT_VIEW_FIELDS_FRAGMENT } from './fragments';
 
 const FIELD_LABEL_PROPS = { fontSize: 16, fontWeight: 700 };
 
+const AGREEMENT_MUTATION_FIELDS_FRAGMENT = gql`
+  fragment AgreementMutationFields on Agreement {
+    id
+    ...AgreementViewFields
+    account {
+      id
+      ... on AccountWithHost {
+        # Refetch account agreements count to update the cache
+        hostAgreements {
+          totalCount
+        }
```

Comment:
This will make sure agreements are reflected in the expenses list as soon as they're added/removed (no page refresh needed).